examples/tutorials/hybrid_demucs_tutorial.py

```diff
@@ -0,0 +1,360 @@
+# internal extra training data.
+# This specific model is suited for higher sample rates, around 44.1 kHZ
+# and has a nfft value of 4096 with a depth of 6 in the model implementation.
+
+from torchaudio.prototype import pipelines
+bundle = pipelines.HDEMUCS_HIGH_MUSDB_PLUS
+
+model = bundle.get_model()
+
+sample_rate = bundle.sample_rate
+
+print(f"Sample rate: {sample_rate}")
+
+
+######################################################################
+# 4. Configure the application function
+# -------------------------------------
+# 
+# Due to the nature of the model, it is very difficult to have sufficient
+# memory to apply the model to  an entire song at once. As a result,
+# to obtain the separated sources of a full
+# song, the song must be chunked into smaller segments and run through the
+# model piece by piece, and then rearranged back together.
+# 
+# When doing this, it is important to ensure some
+# overlap between each of the chunks, to accommodate for artifacts at the
+# edges. Due to the nature of the model, sometimes the edges have
+# inaccurate or undesired sounds included.
+# 
+# We provide a sample implementation of chunking and arrangement below. This
+# implementation takes an overlap of 1 second on each side, and then does
+# a linear fade in and fade out on each side. Using the faded overlaps, I
+# add these segments together, to ensure a constant volume throughout.
+# This accommodates for the artifacts by using less of the edges of the
+# model outputs.
+# 
+
+from torch.nn import functional as F
+from torchaudio.transforms import Fade
+
+def apply_model(model, mix, segment=10.,
+                overlap=0.1, device=None,
+                ):
+    """
+    Apply model to a given mixture. Use fade, and add segments together in order to add model segment by segment. 
+
+    Args:
+        segment (int): segment length in seconds
+        device (torch.device, str, or None): if provided, device on which to
+            execute the computation, otherwise `mix.device` is assumed.
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+    """
+    if device is None:
+        device = mix.device
+    else:
+        device = torch.device(device)
+
+    model.to(device)
+    batch, channels, length = mix.shape
+    
+    chunk_len = int(sample_rate * segment * (1 + overlap))
+    prev = 0
+    index = chunk_len
+    fade = Fade(fade_in_len = 0, fade_out_len = int(overlap * sample_rate), fade_shape='linear')
+    
+    final = torch.zeros(batch, len(model.sources), channels, length, device=device)
+    
+    while index < length:
+        chunk = mix[:,:,prev:index]
+        with torch.no_grad():
+            out = model.forward(chunk)
+            
+        if prev != 0:
+            fade = Fade(fade_in_len = int(overlap * sample_rate), fade_out_len = int(overlap * sample_rate), fade_shape='linear')
```

Comment:
This instantiates the transform with same parameter over and over again. should be done once before entering the while loop.